### PR TITLE
Add single file forwarder

### PR DESCRIPTION
**Pull request by yuqi-wu**

### CODE REVIEW

File: src/forwarders/single_file.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+exports.initialise = function (options) {
+    return send.bind(null, options.fwdDir);
+};
+
+function send (directory, data, type, separator, callback) {
+    try {
+        var dataObj = JSON.parse(data);
+        dataObj['log.time'] = new Date().toLocaleString();
+        fs.appendFile(
+            path.join(directory, 'boomcatch-log.' + extensions[type || 'default']),
```
Comment (evisong) [inline src/forwarders/single_file.js]: I guess when appended multiple beacon jsons, the file is no longer a valid JSON, so what about just use `.log` extension?

File: src/forwarders/single_file.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    path = require('path'),
+    extensions;
+
+exports.initialise = function (options) {
+    return send.bind(null, options.fwdDir);
+};
+
+function send (directory, data, type, separator, callback) {
+    try {
+        var dataObj = JSON.parse(data);
```
Comment (evisong) [inline src/forwarders/single_file.js]: The data is only parse-able and stringify-able when the `type` is `'json'`, better keep other types untouched.

File: src/forwarders/single_file.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        dataObj['log.time'] = new Date().toLocaleString();
```
Comment (evisong) [inline src/forwarders/single_file.js]: Is it better to normalize to UTC time? Btw let's use `logTime` as key, `.` is not friendly for JSON manipulations.

File: src/forwarders/single_file.js (hunk 1 above)
Comment (yuqi-wu) [reply src/forwarders/single_file.js]: Make sense. Updated.

File: src/forwarders/single_file.js (hunk 2 above)
Comment (yuqi-wu) [reply src/forwarders/single_file.js]: Cool. Fixed.

File: src/forwarders/single_file.js (hunk 3 above)
Comment (yuqi-wu) [reply src/forwarders/single_file.js]: Both comments are fixed. Thanks.

Comment (evisong) [reply src/forwarders/single_file.js]: https://stackoverflow.com/a/15952652/245345

**Comment by evisong** [comment]
I've no further comments, thanks.

Merging.

**Pull request merged**